README.md

```diff
@@ -1,23 +1,150 @@
-onOffice-SDK
-============
+# onOffice-SDK
 
-This project is intended as a basic client for the onOffice API.
+This project is the official PHP API client for the
+[onOffice API](https://apidoc.onoffice.de/)
+supported by the onOffice GmbH.
 
-It contains code for the proper generation of a request with a given token, secret, request action
-and parameters.
-There is no information about API actions and their proper parameters included in
-this package.
+* **The HTTP protocol** is used to communicate with the API.
+* An **Access Token** is used to ensure a **secure** communication with the API.
+* The intention is to have **lightweight** client that can be used in other environments
 
+**Table of Contents**
+* [Quickstart Example](#quickstart-example)
+* [Usage](#usage)
+  * [Client](#client)
+  * [Parameters](#parameters)
+  * [Request](#request)
+  * [Response](#respone)
+* [API Documentation](#api-documentation)
+* [Contributing](#contributing)
+* [License](#license)
 
-Contributing
-------------
+## Quickstart Example
+
+```php
+$pSDK = new onOfficeSDK();
+$pSDK->setApiVersion('latest');
+
+$parametersReadEstate = [
+	'data' => [
+		'Id',
+		'kaufpreis',
+		'lage',
+	],
+	'listlimit' => 10,
+	'sortby' => [
+		'kaufpreis' => 'ASC',
+		'warmmiete' => 'ASC',
+	],
+	'filter' => [
+		'kaufpreis' => [
+			['op' => '>', 'val' => 300000],
+		],
+		'status' => [
+			['op' => '=', 'val' => 1],
+		],
+	],
+];
+
+$handleReadEstate = $pSDK->callGeneric(onOfficeSDK::ACTION_ID_READ, 'estate', $parametersReadEstate);
+
+$pSDK->sendRequests('put the token here', 'and secret here');
+
+var_export($pSDK->getResponseArray($handleReadEstate));
+```
+
+Checkout the [example](/example.php) to see a possible implementation of
+this client.
+
+## Usage
+
+### Client
+
+The `onOfficeSDK` is responsible for creating HTTP Requests and
+receiving HTTP Responses from the official API
+
+```php
+$pSDK = new onOfficeSDK();
+$pSDK->setApiVersion('latest');
+```
+
+Make sure that the correct API version is used for your client.
+By default this value is set to `latest`.
+
+### Parameters
+
+The parameters are transferred as JSON in the HTTP Request.
+The client uses the official
+[PHP array notation](https://www.php.net/manual/en/book.json.php)
+before transforming the array to JSON.
+
+```php
+$parametersReadEstate = [
+	'data' => [
+		'Id',
+		'kaufpreis',
+		'lage',
+	],
+	'listlimit' => 10,
+	'sortby' => [
+		'kaufpreis' => 'ASC',
+		'warmmiete' => 'ASC',
+	],
+	'filter' => [
+		'kaufpreis' => [
+			['op' => '>', 'val' => 300000],
+		],
+		'status' => [
+			['op' => '=', 'val' => 1],
+		],
+	],
+];
+```
+
+### Request
+
+To create a request to the API a an `ACTION_ID` is needed.
+In the class `onOfficeSDK` are several constants defined,
+that can be used, so there is no need to copy these `ACTION_IDs`. 
+
+To send the request an API token and API secret is needed,
+checkout the [official API documentation](#api-documentation)
+how to get these credentials.

```

Comment:
I don't think that this sentence is quite correct. Maybe use something more like this?

> A **token** and **secret** are needed to send a request to the API. Check out the [official API documentation](#api-documentation) for information on how to aquire these credentials.